### PR TITLE
non-monotonic sums have the /gauge suffix added in GMP exporter

### PR DESCRIPTION
**Pull request by dashpole**
Fixes https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/issues/675

### CODE REVIEW

File: exporter/collector/googlemanagedprometheus/naming.go [hunk 1]
```diff
@@ -28,7 +28,11 @@ func GetMetricName(baseName string, metric pmetric.Metric) (string, error) {
 	// Second, ad the GMP-specific suffix
 	switch metric.Type() {
 	case pmetric.MetricTypeSum:
-		return compliantName + getUnknownMetricSuffix(metric.Sum().DataPoints(), "/counter", "counter"), nil
+		if metric.Sum().IsMonotonic() {
+			return compliantName + getUnknownMetricSuffix(metric.Sum().DataPoints(), "/counter", "counter"), nil
+		}
+		// Non-monotonic sums are converted to GCM gauges
+		return compliantName + "/gauge", nil
```
Comment (aabmass) [inline exporter/collector/googlemanagedprometheus/naming.go:35]: nit i'd consider monotonic the happy path and would swap which case ends up in the indented if block

**Review by aabmass — APPROVED** [review]
Nit consider adding one of these to the GMP test fixture

File: exporter/collector/googlemanagedprometheus/naming.go (hunk 1 above)
Comment (dashpole) [reply exporter/collector/googlemanagedprometheus/naming.go]: done

**Comment by dashpole** [comment]
I'm adding that case in https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/pull/674

File: exporter/collector/googlemanagedprometheus/naming.go [hunk 2]
```diff
@@ -28,6 +28,10 @@ func GetMetricName(baseName string, metric pmetric.Metric) (string, error) {
 	// Second, ad the GMP-specific suffix
 	switch metric.Type() {
 	case pmetric.MetricTypeSum:
+		if !metric.Sum().IsMonotonic() {
+			// Non-monotonic sums are converted to GCM gauges
+			return compliantName + "/gauge", nil
```
Comment (aabmass) [inline exporter/collector/googlemanagedprometheus/naming.go:33]: Sorry one more question, why doesn't this case use `getUnknownMetricSuffix`?

Comment (dashpole) [reply exporter/collector/googlemanagedprometheus/naming.go:33]: Prometheus unknown is always an OTel gauge.  When we "double-write" this gauge, we copy the gauge into a monotonic counter as well, hence the handling above.  There isn't a way to get a non-monotonic sum from an unknown metric point.

**Review by damemi — APPROVED** [review]

**Pull request merged**